akka-stream/src/main/scala/akka/stream/javadsl/Flow.scala

```diff
@@ -319,7 +319,35 @@ final class Flow[-In, +Out, +Mat](delegate: scaladsl.Flow[In, Out, Mat]) extends
    * '''Cancels when''' downstream cancels
    */
   def mapConcat[T](f: function.Function[Out, java.lang.Iterable[T]]): javadsl.Flow[In, T, Mat] =
-    new Flow(delegate.mapConcat { elem ⇒ Util.immutableSeq(f(elem)) })
+    new Flow(delegate.statefulMapConcat { () ⇒ elem ⇒ Util.immutableSeq(f(elem)) })
+
+  /**
+    * Transform each input element into an `Iterable` of output elements that is
+    * then flattened into the output stream. Transformation is stateful
+    * meaning it's taking into account previous messages (hysteresis).
```

Comment:
The second sentence is a bit too indirect (i.e. where is the state?). What about:

> The transformation is meant to be stateful, which is enabled by creating the transformation function anew for every materialization—the returned function will typically close over mutable objects to store state between invocations. For the stateless variant see `mapConcat`.